src/ProductConstructionService/SubscriptionTriggerer/SubscriptionTriggererConfiguration.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.DotNet.Kusto;
+using Azure.Storage.Queues;
+using Azure.Identity;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.DotNet.Internal.Logging;
+
+namespace SubscriptionTriggerer;
+public class SubscriptionTriggererConfiguration
+{
+    private const string ApplicationInsightsConnectionString = "APPLICATIONINSIGHTS_CONNECTION_STRING";
+    private const string AspnetcoreEnvironment = "ASPNETCORE_ENVIRONMENT";
+    private const string QueueConnectionString = "QueueConnectionString";
+    private const string ManagedIdentityClientId = "ManagedIdentityClientId";
+    private const string SqlConnectionStringUserIdPlaceholder = "USER_ID_PLACEHOLDER";
+    private const string DatabaseConnectionString = "BuildAssetRegistrySqlConnectionString";
+
+    public static IServiceCollection RegisterServices(IServiceCollection services, ITelemetryChannel telemetryChannel)
+    {
+        RegisterLogging(services, telemetryChannel);
+
+        IConfiguration config = GetConfiguration();
+        services.AddSingleton(_ => config);
+
+        string databaseConnectionString = config.GetRequiredValue(DatabaseConnectionString)
+            .Replace(SqlConnectionStringUserIdPlaceholder, config[ManagedIdentityClientId]);
+
+        services.AddBuildAssetRegistry((provider, options) =>
+        {
+            options.UseSqlServerWithRetry(databaseConnectionString);
+        });
+
+        services.Configure<OperationManagerOptions>(o => { });
+        services.AddTransient<OperationManager>();
+
+        services.AddTransient<IHostEnvironment>(_ => new HostEnvironment());
+        services.AddTransient<DarcRemoteMemoryCache>();
+        services.AddTransient<IProcessManager>(sp => ActivatorUtilities.CreateInstance<ProcessManager>(sp, "git"));
+        services.AddTransient<IVersionDetailsParser, VersionDetailsParser>();
+        services.AddTransient<IBasicBarClient, SqlBarClient>();
+        services.AddTransient(_ => new QueueClient(
+            new Uri(config.GetRequiredValue(QueueConnectionString)),
+            new DefaultAzureCredential(new DefaultAzureCredentialOptions
+            {
+                ManagedIdentityClientId = config[ManagedIdentityClientId]
+            })));
+        services.AddKustoClientProvider("Kusto");
+
+        return services;
+    }
+
+    private static IConfiguration GetConfiguration()
+    {
+        var environment = Environment.GetEnvironmentVariable(AspnetcoreEnvironment);
+        var builder = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.json", optional: false)
+            .AddJsonFile($"appsettings.{environment}.json", optional: true);
+        return builder.Build();
+    }
```

Comment:
This could go away if we used an app builder

Reply:
it's gone now